projects/external-components/src/lib/primeng-selectors/components/generate-with-ai/generate-with-ai.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, OnInit, ElementRef, ViewChild, AfterViewInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { CommonExternalComponent } from '../common-external/common-external.component';
+import { Observable } from 'rxjs';
+import { Clipboard } from '@angular/cdk/clipboard';
+
+
+
+interface MessagePart {
+  type: 'text' | 'code';
+  content: string;
+}
+
+interface Message {
+  isUser: boolean;
+  parts: MessagePart[];
+}
+
+@Component({
+  selector: 'hmi-ext-generate-with-ai',
+  templateUrl: './generate-with-ai.component.html',
+  styleUrls: ['./generate-with-ai.component.css']
+
+})
+export class GenerateWithAiComponent extends CommonExternalComponent implements OnInit, AfterViewInit, OnDestroy {
+  messages: Message[] = [];
+
+  @ViewChild('messagesContainer') private messagesContainer!: ElementRef;
+  componentName: string = "";
+  messageData: any = {}; // Initialize messageData
+  actions:any;
+  response$!: Observable<any>
+  content: any;
+
+  constructor(private route: ActivatedRoute, private cdr: ChangeDetectorRef, private clipboard: Clipboard) {
+    super();
+   }
+
+   ngOnInit(): void {
+    
+    this.fieldObj.value = { newMessage: '' };   
+  }
+
+  ngAfterViewInit() {
+    this.scrollToBottom();
+  }
+
+  sendMessage() {
+    this.messageData.newMessage = this.fieldObj.value.newMessage; 
+    if (this.fieldObj.value.newMessage.trim()) {
+      this.messages.push({
+        isUser: true,
+        parts: [{ type: 'text', content: this.fieldObj.value.newMessage.trim() }]
+      });
+      this.fieldObj.value.newMessage = '';      
+      this.initializeEvents.emit({ name: "fireEvent", "events": [this.fieldObj.events[0]], data: null});
```

Comment:
apply a condition to get the sendmessage event only otherwise don't fire an event.